File: templates/docker/Dockerfile.java
FROM maven:3.8-openjdk-17 AS builder
WORKDIR /app
COPY pom.xml .
# ...existing code...
RUN mvn -B -DskipTests=true dependency:go-offline

COPY . .
RUN mvn -B -DskipTests=true package

FROM eclipse-temurin:17-jre
WORKDIR /app
COPY --from=builder /app/target/*.jar /app/app.jar
ENV JAVA_OPTS=""
ENTRYPOINT ["sh","-c","java $JAVA_OPTS -jar /app/app.jar"]

File: templates/docker/Dockerfile.go
FROM golang:1.20-alpine AS builder
WORKDIR /app
ENV CGO_ENABLED=0
COPY go.mod go.sum ./
RUN go mod download
COPY . .
RUN go build -ldflags="-s -w" -o /app/bin/app ./...

FROM alpine:3.18
RUN addgroup -S app && adduser -S -G app app
COPY --from=builder /app/bin/app /usr/local/bin/app
USER app
ENTRYPOINT ["/usr/local/bin/app"]

